ui/component/or-attribute-card/src/index.ts

```diff
@@ -0,0 +1,441 @@
+                return this.mainValueLastPeriod;
+            });
+
+        Promise.all([p1, p2])
+            .then((returnvalues) => {
+                this.delta = this.getFormattedDelta(returnvalues[0], returnvalues[1]);
+            });
+
+    }
+
+    protected sanitiseDataPoints(data: ValueDatapoint<any>[]): ValueDatapoint<any>[] {
+
+        // if there's no measurement for the first data point in time, assume 0
+        if (data[0] && !data[0].y) {
+            data[0].y = 0;
+        }
+
+        // if there's no measurement for the last data point, use the highest available
+        if (data[(data.length - 1)] && !data[(data.length - 1)].y) {
+            data[(data.length - 1)].y = this.getHighestValue(data);
+        }
+
+        return data;
+    }
+
+    protected getTotalValue(data: ValueDatapoint<any>[]): number {
+        return data.reduce(( acc: number, val: ValueDatapoint<any> ) => {
+            return val.y ? acc + Math.round(val.y) : acc;
+        }, 0);
+    }
+
+    protected getHighestValue(data: ValueDatapoint<any>[]): number {
+        return Math.max.apply(Math, data.map((e: ValueDatapoint<any>) => e.y || false ));
+    }
+
+    protected getFormattedValue(value: number): {value: number, unit: string, formattedValue: string} {
+        const format = getMetaValue(MetaItemType.FORMAT, this.asset.attributes![this.attributeName], undefined);
+        const unit = format.split(" ").pop();
+        return {
+            value: value,
+            unit: unit,
+            formattedValue: i18next.t(format, { postProcess: "sprintf", sprintf: [value] }).trim()
+        };
+    }
+
+    protected getFormattedDelta(currentPeriodVal: number, lastPeriodVal: number): {val?: number, unit?: string} {
+        if (currentPeriodVal && lastPeriodVal) {
+            if (lastPeriodVal === 0 && currentPeriodVal === 0) {
+                return {val: 0, unit: "%"};
+            } else if (lastPeriodVal === 0 && currentPeriodVal !== 0) {
+                return {val: 100, unit: "%"};
+            } else {
+                const math = Math.round((currentPeriodVal - lastPeriodVal) / lastPeriodVal * 100);
+                return {val: math, unit: "%"};
+            }
+        } else {
+            return {};
+        }
+    }
+
+    protected _getPeriodOptions() {
+        return [
+            {
+                text: i18next.t(DatapointInterval.HOUR),
+                value: "hour"
+            },
+            {
+                text: i18next.t(DatapointInterval.DAY),
+                value: "day"
+            },
+            {
+                text: i18next.t(DatapointInterval.WEEK),
+                value: "week"
+            },
+            {
+                text: i18next.t(DatapointInterval.MONTH),
+                value: "month"
+            },
+            {
+                text: i18next.t(DatapointInterval.YEAR),
+                value: "year"
+            }
```

Comment:
You can just pass value through `i18next.t()` at the point at which these options are consumed so just return desired `DatapointInterval` options here

Reply:
I've pushed a solution, hope it's what you meant!